stdlib/public/core/SIMDVectorTypes.swift.gyb

```diff
@@ -185,6 +187,34 @@ extension SIMD${n} where Scalar: BinaryFloatingPoint {
   }
 }
 
+extension SIMD${n} : AdditiveArithmetic where Scalar : FloatingPoint {}
+
+extension SIMD${n} : VectorNumeric where Scalar : FloatingPoint {}
+
+extension SIMD${n} : Differentiable where Scalar : Differentiable & BinaryFloatingPoint {
+  public typealias TangentVector = SIMD${n}
+  public typealias CotangentVector = SIMD${n}
+  public typealias AllDifferentiableVariables = SIMD${n}
+  public func tangentVector(from cotangent: CotangentVector) -> TangentVector {
+    return cotangent
+  }
+}
+
+extension SIMD${n}
+  where Scalar : Differentiable & BinaryFloatingPoint,
+        Scalar.CotangentVector : BinaryFloatingPoint {
+  public func _vjpSubscript(index: Int) ->
+  (Scalar, (Scalar.CotangentVector) -> CotangentVector)
```

Comment:
Indent by 2 and put the `->` on this line.
```suggestion
  -> (Scalar, (Scalar.CotangentVector) -> CotangentVector)
```

Reply:
To verify:
```swift
public func _vjpSubscript(index: Int)
  -> (Scalar, (Scalar.CotangentVector) -> CotangentVector)
```

Reply:
Yes. VJP functions should not be public though. When the original function is public, a VJP function should be internal. When it's internal, if the original function has `@inlinable`, the VJP function should be `@inlinable`, otherwise the VJP function should be `@usableFromInline`.

Reply:
Ahh okay. But for let's say `_vjpSum`, the original function (in the protocol) is marked as public and `@inlinable`. How should `_vjpSum` be marked? Looking at `Tensor`, its `_vjpSum` is just `@inlinable` in a regular extension

Reply:
If the original function is `@inlinable`, the VJP should also be `@inlinable`. In all cases where the original function is public, the VJP should be internal.

Reply:
But for `SIMD.sum()`, it's not `@inlinable`, so the VJP should be `@usableFromInline`.

Reply:
Ahh okay thanks!